src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs

```diff
@@ -905,6 +905,553 @@ public void MatchExceptionHandlers()
+";
+
+            var src2 = @"
+if (!(o1 is int i) && (y == 7)) return;
+if (!(a is int s)) return;
+if (!(b is string t1)) return;
+if (!(c is int)) return;
+";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs {
+                { "if (!(o is int i) && (y == 7)) return;", "if (!(o1 is int i) && (y == 7)) return;" },
+                { "i", "i" },
+                { "return;", "return;" },
+                { "if (!(a is string s)) return;", "if (!(a is int s)) return;" },
+                { "s", "s" },
+                { "return;", "return;" },
+                { "if (!(b is string t)) return;", "if (!(b is string t1)) return;" },
+                { "t", "t1" },
+                { "return;", "return;" },
+                { "if (!(c is int j)) return;", "if (!(c is int)) return;" },
+                { "return;", "return;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchCasePattern_UpdateInsert()
+        {
+            var src1 = @"
+switch(shape)
+{
+    case Circle c: return 1;
+    default: return 4;
+}
+";
+
+            var src2 = @"
+switch(shape)
+{
+    case Circle c1: return 1;
+    case Point p: return 0;
+    default: return 4;
+}
+";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs {
+                { "switch(shape) {     case Circle c: return 1;     default: return 4; }", "switch(shape) {     case Circle c1: return 1;     case Point p: return 0;     default: return 4; }" },
+                { "case Circle c: return 1;", "case Circle c1: return 1;" },
+                { "case Circle c:", "case Circle c1:" },
+                { "c", "c1" },
+                { "return 1;", "return 1;" },
+                { "default: return 4;", "default: return 4;" },
+                { "return 4;", "return 4;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchCasePattern_UpdateDelete()
+        {
+            var src1 = @"
+switch(shape)
+{
+    case Point p: return 0;
+    case Circle c: return 1;
+    default: return 4;
+}
+";
+
+            var src2 = @"
+switch(shape)
+{
+    case Circle c1: return 1;
+    
+    default: return 4;
+}
+";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs {
+                { "switch(shape) {     case Point p: return 0;     case Circle c: return 1;     default: return 4; }", "switch(shape) {     case Circle c1: return 1;          default: return 4; }" },
+                { "case Point p: return 0;", "case Circle c1: return 1;" },
+                { "case Point p:", "case Circle c1:" },
```

Comment:
>case Point p:", "case Circle c1:" }, [](start = 19, length = 36)

Also odd. I'd expect "case Circle c" to match better  with "case Circle c1" than "case Point p" does.